x-pack/plugins/security_solution/public/detections/pages/detection_engine/detection_engine_no_signal_index.tsx

```diff
@@ -10,19 +10,31 @@ import { EmptyPage } from '../../../common/components/empty_page';
 import * as i18n from './translations';
 import { useKibana } from '../../../common/lib/kibana';
 
-export const DetectionEngineNoIndex = React.memo(() => {
+const DetectionEngineNoIndexComponent: React.FC<{
+  needsListsIndex: boolean;
+  needsSignalsIndex: boolean;
+}> = ({ needsListsIndex, needsSignalsIndex }) => {
   const docLinks = useKibana().services.docLinks;
+  let context: string = '';
+  if (needsSignalsIndex && needsListsIndex) {
+    context = i18n.NEEDS_SIGNALS_AND_LISTS_INDEXES;
+  } else if (needsSignalsIndex) {
+    context = i18n.NEEDS_SIGNALS_INDEX;
+  } else if (needsListsIndex) {
+    context = i18n.NEEDS_LISTS_INDEXES;
+  }
+
   return (
     <EmptyPage
       actionPrimaryIcon="documents"
       actionPrimaryLabel={i18n.GO_TO_DOCUMENTATION}
       actionPrimaryUrl={`${docLinks.ELASTIC_WEBSITE_URL}guide/en/security/${docLinks.DOC_LINK_VERSION}/detection-engine-overview.html#detections-permissions`}
       actionPrimaryTarget="_blank"
-      message={i18n.NO_INDEX_MSG_BODY}
+      message={i18n.NEEDS_INDEX_PERMISSIONS(context)}
       data-test-subj="no_index"
       title={i18n.NO_INDEX_TITLE}
     />
   );
-});
+};
 
-DetectionEngineNoIndex.displayName = 'DetectionEngineNoIndex';
```

Comment:
In testing this behavior I found that setting this displayName had no effect in React Dev Tools; the displayName was already inferred.

Reply:
I believe this is a result of declaring the component as a `React.FC<{}>` -- similarly this can be auto-set using the named function syntax as well.